File: test/domain/data/japanese_prefectures_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:libcheck/domain/data/japanese_prefectures.dart';

void main() {
  group('JapanesePrefectures', () {
    test('has 7 region groups', () {
      expect(JapanesePrefectures.regions.length, 7);
    });

    test('contains exactly 47 prefectures in total', () {
      expect(JapanesePrefectures.allPrefectures.length, 47);
    });

    test('has no duplicate prefectures', () {
      final all = JapanesePrefectures.allPrefectures;
      expect(all.toSet().length, all.length);
    });
  });
}

File: src/domain/data/japanesePrefectures.test.ts
import { describe, expect, test } from 'vitest';

import {
  JAPANESE_PREFECTURE_REGIONS,
  allPrefectures,
} from '@/domain/data/japanesePrefectures';

describe('JapanesePrefectures', () => {
  test('has 7 region groups', () => {
    expect(JAPANESE_PREFECTURE_REGIONS.length).toBe(7);
  });

  test('contains exactly 47 prefectures in total', () => {
    expect(allPrefectures.length).toBe(47);
  });

  test('has no duplicate prefectures', () => {
    const all = allPrefectures;
    expect(new Set(all).size).toBe(all.length);
  });
});
